pkg/controller/cluster/http_handlers.go

```diff
@@ -212,7 +212,7 @@ func (c *Controller) CRDConversionHandler(w http.ResponseWriter, r *http.Request
 		cr := unstructured.Unstructured{}
 		if err := cr.UnmarshalJSON(obj.Raw); err != nil {
 			log.Println(err)
-			w.WriteHeader(500)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
 			req.Response.Result.Status = metav1.StatusFailure
 			rawResp, _ := json.Marshal(req)
 			if _, err = w.Write(rawResp); err != nil {
```

Comment:
once you have written the body with `http.Error` unfortunately you cannot further write the `rawResp` in HTTP that's why I only suggested one change not all.

Reply:
oh shiet, the k8s api wants me to return their format with status failure instead, I'll revert